pkg/build/builder/docker.go

```diff
@@ -94,6 +101,12 @@ func (d *DockerBuilder) Build() error {
 
 	cname := containerName("docker", d.build.Name, d.build.Namespace, "post-commit")
 	if err := execPostCommitHook(d.dockerClient, d.build.Spec.PostCommit, buildTag, cname); err != nil {
+		updateBuildFailureReason(d.build, api.StatusReasonPostCommitHookFailed)
+		if _, updateErr := d.client.UpdateDetails(d.build); updateErr != nil {
+			glog.V(0).Infof("error: An error occured while updating the build status %v", updateErr)
+			// NOTE: handle me!
```

Comment:
the s2ibuilder doesn't have an event recorder, the event recording (for builds at least) are mostly done in the controller, should i attach one to it? in that case i need to attach one to the docker builder as well.

>if it's a conflict error, refetching and retrying the update is reasonable. otherwise no, just fail.

i think this is what the intended purpose of the validation is, to avoid conflicting updates. at least for revision.